src/embind/embind.js

```diff
@@ -2058,21 +2095,9 @@ var LibraryEmbind = {
         };
 
         whenDependentTypesAreResolved([], rawArgTypes, function(argTypes) {
-            classType.registeredClass.constructor_body[argCount - 1] = function constructor_body() {
-                if (arguments.length !== argCount - 1) {
-                    throwBindingError(humanName + ' called with ' + arguments.length + ' arguments, expected ' + (argCount-1));
-                }
-                destructors.length = 0;
-                args.length = argCount;
-                for (var i = 1; i < argCount; ++i) {
-                    args[i] = argTypes[i]['toWireType'](destructors, arguments[i - 1]);
-                }
-
-                var ptr = invoker.apply(null, args);
-                runDestructors(destructors);
-
-                return argTypes[0]['fromWireType'](ptr);
-            };
+            // Insert empty slot for context type (argTypes[1]).
+            argTypes.splice(1, 0, null);
+            classType.registeredClass.constructor_body[argCount - 1] = craftInvokerFunction(humanName, argTypes, null, invoker, rawConstructor);
             return [];
```

Comment:
Is this part of the PR related? If so, how?

Reply:
Unlike all the other function wrappers, this one for some reason didn't use `craftInvokerFunction` helper, so when I added async finalizers to that helper, class constructors would still return invalid instances instead of Promises.

This change is both a simplification to reuse the same helper, and allows class constructors to be async as well.

Reply:
I see, thanks. sgtm